### PR TITLE
Fix id_fields issue

### PR DESCRIPTION
**Pull request by afrancis13**
I was working with bungiesearch yesterday and I noticed that I was having a lot of trouble with models that had defined primary keys (models in which the pk field was automatically generated as 'id'). So I went through the code and wrote a few tests and realized that bungiesearch wouldn't pass the tests. I included the additional tests (you can move one commit backwards in Travis to see them failing) so you can see the issue. Please let me know if you have any questions. Also, if you confirm this makes sense, and that there isn't a simple workaround that I'm missing, let me know if I should include or get rid of the additional testing, at which point I'll push and fixup the commits.

### CODE REVIEW

File: tests/core/test_bungiesearch.py [hunk 1]
```diff
@@ -100,6 +137,9 @@ def test_no_results(self):
         '''
         self.assertEqual(list(Article.objects.search.query('match', _all='nothing')), [], 'Searching for "nothing" did not return an empty list on iterator call.')
         self.assertEqual(Article.objects.search.query('match', _all='nothing')[:10], [], 'Searching for "nothing" did not return an empty list on get item call.')
+        
+        self.assertEqual(list(User.objects.search.query('match', _all='nothing')), [], 'Searching for "nothing" did not return an empty list on iterator call.')
+        self.assertEqual(list(User.objects.search.query('match', _all='nothing')), [], 'Searching for "nothing" did not return an empty list on iterator call.')
```
Comment (ChristopherRabotin) [inline tests/core/test_bungiesearch.py]: Line 142 is identical to line 141. You want to test the list (`__iter__` function) and the slice (`__getitem__` function) operations here so one should be `list(...)` and the other `()[x:y]` (cf. lines 138 and 139).

**Comment by ChristopherRabotin** [comment]
Good stuff! I hadn't had actually noticed that problem since we've only used `id` as a primary key. There are a few styles changes I'll make so I'll merge it in for now, do some changes on another branch and get your input prior to merging those changes in.

**Comment by afrancis13** [comment]
Great! Another thing I noticed when I was going through this is that if your primary key is a field that's a relation (i.e. models.OneToOneField, see example in Django docs below), bungiesearch won't work. Is there a reason you're filtering relation fields out in the _get_fields method in indices.py?
https://docs.djangoproject.com/en/1.8/topics/db/examples/one_to_one/

**Comment by ChristopherRabotin** [comment]
@afrancis13 Yes, it isn't supported because I wasn't sure how to handle potential conflicts. Even if Django has that One to One relation, I worried that any table rows added manually may lead to an erroneous index.

Any thoughts on implementing this?

**Comment by afrancis13** [comment]
@ChristopherRabotin Thanks for the response, I'll look into this when I get the chance.

**Pull request merged**